src/utils/todo/todoFilter.js

```diff
@@ -0,0 +1,78 @@
+import { FILTER_STATE } from '../../const/todo.const.js';
+
+import { renderList, updateSummary } from '../../ui/todo/todo.render.js';
+import { changeFilterClass } from '../../ui/todo/todo.dom.js';
+import { deleteCompleteTodo } from '../../store/todo.store.js';
+
+export class TodoFilter {
+  filterState = FILTER_STATE.ALL;
+
+  showCompleteTodo() {
+    const filteredTodo = this.todos.filter((todo) => todo.complete);
+
+    renderList.bind(this)(filteredTodo);
+    this.filterState = FILTER_STATE.COMPLETE;
+  }
+
+  showActiveTodo() {
+    const filteredTodo = this.todos.filter((todo) => !todo.complete);
+
+    renderList.bind(this)(filteredTodo);
+    this.filterState = FILTER_STATE.ACTIVE;
+  }
+
+  showAllTodo() {
+    renderList.bind(this)(this.todos);
+
+    this.filterState = FILTER_STATE.ALL;
+  }
+
+  updateFilter() {
+    if (this.filterState == FILTER_STATE.ACTIVE) {
+      this.showActiveTodo();
+    } else if (this.filterState == FILTER_STATE.COMPLETE) {
+      this.showCompleteTodo();
+    }
+  }
+
+  addDeleteEvent() {
+    this.$btnClear.addEventListener('click', () => {
+      deleteCompleteTodo.bind(this)();
+      updateSummary.bind(this)();
+    });
+  }
+
+  changeFilter() {
+    this.$filter.addEventListener('click', (e) => {
+      const $target = e.target;
+
+      if (!this.todos || this.todos.length < 1) return;
+
+      if ($target.classList.contains('btn-todo-active')) {
+        this.showActiveTodo();
+      } else if ($target.classList.contains('btn-todo-completed')) {
+        this.showCompleteTodo();
+      } else if ($target.classList.contains('btn-todo-all')) {
+        this.showAllTodo();
+      }
```

Comment:
p5; dataset으로 체크해주는 것도 괜찮을거 같아요! 
<button class="btn-todo-filter btn-todo-all is-active" data-filter="all">All</button>
 ... 

```suggestion
      const handlers = {
          [FILTER_STATE.ALL]: () => this.showAllTodo(),
          [FILTER_STATE.ACTIVE]: () => this.showActiveTodo(),
          [FILTER_STATE.COMPLETE]: () => this.showCompleteTodo(),
        };
        
        const filterType = dataset.filter;
        handlers[filterType]?.();
```